io-sim/src/Control/Monad/IOSim.hs

```diff
@@ -777,6 +779,18 @@ schedule thread@Thread{
                                                (CatchFrame handler k ctl) }
       schedule thread' simstate
 
+    Evaluate expr k -> do
+      mbWHNF <- unsafeIOToST $ try $ evaluate expr
```

Comment:
I'm disappointed by `unsafeIOToST` here -- is there a safe alternative?

Reply:
I don't think there's a way to avoid `unsafe` function when we want to use `evaluate` (which, I think, is a bit magical how it deals with exceptions).

Reply:
I agree :( thanks for confirming.